### PR TITLE
Fix ref name to tag to use short ref on blockchain compatibility test

### PR DESCRIPTION
**Pull request by chainchad**
To fix issues like https://github.com/smartcontractkit/chainlink/actions/runs/9776215954/job/26988909636#step:4:1448

### CODE REVIEW

File: .github/workflows/client-compatibility-tests.yml [hunk 1]
```diff
@@ -206,7 +206,7 @@ jobs:
             echo "chainlink_version=$chainlink_version" >> $GITHUB_OUTPUT
           elif [ "$GITHUB_REF_TYPE" = "tag" ]; then
             echo "Fetching Chainlink version from tag"
-            chainlink_version="${{ github.ref }}"
+            chainlink_version="${{ github.ref_name }}"
```
Comment (chainchad) [inline .github/workflows/client-compatibility-tests.yml:209]: Ignore whitespace/formatting, this is the actual fix :)

**Comment by chainchad** [comment]
Going to look at having this re-use an existing image instead of re-building, at least in the case of tagged images.

edit: looks like it tries to do this but doesn't work for actual v* tags.
edit: making it skip the build if the ref is a git tag.

**Comment by chainchad** [comment]
@Tofel what's the best way to test to confirm this is valid?

File: .github/workflows/client-compatibility-tests.yml [hunk 2]
```diff
@@ -267,7 +267,14 @@ jobs:
           page_size: ${{matrix.mirror.page_size}}
 
   build-chainlink:
-    if: always() && needs.should-run.outputs.should_run == 'true' && (needs.select-versions.outputs.evm_implementations != '' || github.event.inputs.base64TestList != '')
+    if: |
+      always() &&
+      needs.should-run.outputs.should_run == 'true' &&
+      github.ref_type != 'tag' &&
```
Comment (chainchad) [inline .github/workflows/client-compatibility-tests.yml:273]: No need to build again if the ref is a tag.

**Review by HenryNguyen5 — APPROVED** [review]

**Review by Tofel — APPROVED** [review]

**Pull request merged**